### PR TITLE
Add clang tidy

### PR DESCRIPTION
**Pull request by henrygerardmoore**
This adds clang tidy to our CI, but not to pre-commit, though it does add a convenience script for running clang-tidy locally. It'll only run on changed files, but it's still pretty slow.

I changed that one cpp file just to make sure that clang-tidy is working properly. Ideally we could apply all clang tidy changes across the code base at once, but that was a big burden that I didn't want to take on, so I think making it required for any files a PR touches is a good middle ground that will ensure the code quality goes up over time.

The `.clang-tidy` is one I've used in personal and work projects before and has a configuration I like. I think that's sufficient for this repo.

### CODE REVIEW

File: Jenkinsfile
Comment (henrygerardmoore) [inline Jenkinsfile:1]: We're not using this.

File: .github/workflows/ros2.yml
Comment (henrygerardmoore) [inline .github/workflows/ros2.yml:1]: Our CI replaces industrial CI

File: Dockerfile [hunk 1]
```diff
@@ -20,6 +20,7 @@ WORKDIR /colcon_ws
 # hadolint ignore=SC1091
 RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
     --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    --mount=type=cache,target=/$HOME/.cache,sharing=locked \
```
Comment (henrygerardmoore) [inline Dockerfile:23]: I think pip in the docker container needs this in CI, was getting errors without it.

File: fuse_core/test/launch_tests/test_parameters.cpp
Comment (henrygerardmoore) [inline fuse_core/test/launch_tests/test_parameters.cpp:1]: All changes in this file are just from clang tidy, wanted to have one changed CPP file to make sure it works.

File: run_clang_tidy.bash
Comment (henrygerardmoore) [inline run_clang_tidy.bash:1]: Just a convenience script I've used before for developers to run clang tidy locally if they should want.

File: .clang-tidy [hunk 2]
```diff
@@ -1,26 +1,29 @@
...
+  -modernize-use-trailing-return-type,
+  -readability-identifier-length,
+  -readability-function-cognitive-complexity,
+  -readability-magic-numbers,
+  -readability-uppercase-literal-suffix,
+  -cppcoreguidelines-pro-type-union-access,
+  -cppcoreguidelines-non-private-member-variables-in-classes,
+  -cppcoreguidelines-pro-type-vararg,
 HeaderFilterRegex: ''
-AnalyzeTemporaryDtors: false
```
Comment (henrygerardmoore) [inline .clang-tidy:23]: No longer a valid option in clang tidy 18

**Review by bgill92 — APPROVED** [review]
I think the changes make sense 👍🏽

**Pull request merged**